### DISCUSSION TITLE
How does MipMap Tail Packing on x360 work.

### DISCUSSION THREAD

**Discussion by TKFRvisionOfficial** [gildor2/UEViewer#267] [Q&A] [answered]
I wanna implement mip map tail unpacking. How do I unpack the mip map tail? How do I get the offset? What are the funcs used here? Those are the question that keep me awake at night. Please Help! Thanks.

**Comment by TKFRvisionOfficial** [comment]
And while where at it. How does repacking work? And why does my dad not come back from the shop...he just wanted to get cigarettes. 

I don't want code if you don't have any I just want to by pointed into the right direction.,.

**Comment by gildor2** [comment, marked as answer]
In UModel, textures are decoded in CTextureData::DecodeXBox360. This does untiling (unswizzling), but doesn't deal with mipmap packing, as UE3 stored all mips in separate arrays.
In XDK documentation, I'd suggest looking at Graphics/Textures folder, especially at "Packed Mipmap Tails" article.

**Reply by TKFRvisionOfficial** [reply]
Aight yeah gonna mark this as the answer. Sadly the xdk docu is pretty shit. Like they have one picture and that's about it. The rest is use these functions from this static library. Thanks anyways. Gonna try my luck with the xenia code.
